logging/impl.go

```diff
@@ -98,14 +100,34 @@ func (imp *impl) Sync() error {
 	return multierr.Combine(errs...)
 }
 
-func (imp *impl) With(args ...interface{}) *zap.SugaredLogger {
-	return imp.AsZap().With(args...)
-}
-
 func (imp *impl) WithOptions(opts ...zap.Option) *zap.SugaredLogger {
 	return imp.AsZap().WithOptions(opts...)
 }
 
+func (imp *impl) WithFields(args ...interface{}) {
+	for keyIdx := 0; keyIdx < len(args); keyIdx += 2 {
+		keyObj := args[keyIdx]
+		var keyStr string
+
+		switch key := keyObj.(type) {
+		case string:
+			keyStr = key
+		case fmt.Stringer:
+			keyStr = key.String()
+		default:
+			continue
+		}
+
+		if keyIdx+1 < len(args) {
+			imp.logFields = append(imp.logFields, zap.Any(keyStr, args[keyIdx+1]))
```

Comment:
A detail that I think has no impact -- if a user calls:
```
logger.WithFields("a", 1)
logger.WithFields("a", 2)
```
I expect this code will add 2 items to the log entry. And I suppose a zap logger would only add `"a": 2`?

I'm sure eyeballing to verify that goutils does not do this is trivial. But might be worth a callout at the top of this function. So a reader knows this was an intentional choice and that "last one wins" isn't considered incorrect.

Reply:
this is a good point still since it does slightly deviate from `zap.SugaredLogger` behavior so I will make sure to document